controllers/front/CartController.php

```diff
@@ -63,6 +63,27 @@ public function init()
         $this->customization_id = (int)Tools::getValue('id_customization');
         $this->qty = abs(Tools::getValue('qty', 1));
         $this->id_address_delivery = (int)Tools::getValue('id_address_delivery');
+
+        if (Tools::getIsset('checkout')) {
+            $product = $this->context->cart->checkQuantities(true);
+            $isValidToCheckout = true;
+
+            if (is_array($product)) {
+                $isValidToCheckout = false;
```

Comment:
Actually, it seems you do not need this variable `$isValidToCheckout` in your code.

if you are in the case `if (is_array($product)) {`, then prepare your error and redirect the customer. :)